<launch>

    <arg name="thename" default="m100"/><!--无人机名称（每台无人机必须不相同）-->
    <arg name="axis" default="2"/><!--坐标系：1为全局自然坐标系，2为局部实验坐标系-->
    <arg name="index" default="8"/>
    <arg name="name" default="uav$(arg index)"/>
    <arg name="num" default="10"/>
    
    <node
        pkg = "cloud_server_communication"
        type = "subscribe"
        name = "$(arg name)_subscribe_node"
        output = "screen"
        respawn = "true" >
        <param name="uav_index" type="int" value="$(arg index)"/>
        <param name="uav_name" type="string" value="$(arg name)"/>
        <param name="axis_id" type="int" value="$(arg axis)"/>
        <param name="num_uav" type="int" value="$(arg num)"/>
        <remap from="vel_cmd" to="vel_list"/>
    </node>

    <node
        pkg = "cloud_server_communication"
        type = "client2ground"
        name = "$(arg name)_client2ground"
        output = "screen"
        respawn = "true">
        <param name="uav_index" type="int" value="$(arg index)"/>
        <param name="uav_name" type="string" value="$(arg name)"/>
        <param name="axis_id" type="int" value="$(arg axis)"/>
        <param name="num_uav" type="int" value="$(arg num)"/>
        <remap from="GPS_position" to="$(arg name)/dji_sdk/gps_position"/>
        <remap from="attitude" to="$(arg name)/dji_sdk/attitude"/>
        <remap from="GPS_health" to="$(arg name)/dji_sdk/gps_health"/>
        <remap from="flight_status" to="$(arg name)/dji_sdk/flight_status"/>
    </node>
<!--
    <node
    pkg = "cloud_server_communication"
    type = "statusUpdate"
    name = "$(arg name)_statusUpdate_node"
    output = "screen"
    respawn = "true">
    <param name="uav_index" type="int" value="$(arg index)"/>
    <param name="uav_name" type="string" value="$(arg name)"/>
    <param name="axis_id" type="int" value="$(arg axis)"/>
    <param name="num_uav" type="int" value="$(arg num)"/>
    </node> -->

</launch>
